File: binding/src/charts/libqvboxplotmodelmapper.cpp
#include <QAbstractItemModel>
#include <QBoxPlotModelMapper>
#include <QBoxPlotSeries>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVBoxPlotModelMapper>
#include <qvboxplotmodelmapper.h>
#include "libqvboxplotmodelmapper.h"
#include "libqvboxplotmodelmapper.hxx"

QVBoxPlotModelMapper* QVBoxPlotModelMapper_new() {
	 return new QVBoxPlotModelMapper();
}

QVBoxPlotModelMapper* QVBoxPlotModelMapper_new2(QObject* parent) {
	 return new QVBoxPlotModelMapper(parent);
}

libqt_string QVBoxPlotModelMapper_Tr(const char* s) {
	QString _ret = QVBoxPlotModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAbstractItemModel* QVBoxPlotModelMapper_Model(const QVBoxPlotModelMapper* self) {
	return self->model();
}

void QVBoxPlotModelMapper_SetModel(QVBoxPlotModelMapper* self, QAbstractItemModel* model) {
	self->setModel(model);
}

QBoxPlotSeries* QVBoxPlotModelMapper_Series(const QVBoxPlotModelMapper* self) {
	return self->series();
}

void QVBoxPlotModelMapper_SetSeries(QVBoxPlotModelMapper* self, QBoxPlotSeries* series) {
	self->setSeries(series);
}

int QVBoxPlotModelMapper_FirstBoxSetColumn(const QVBoxPlotModelMapper* self) {
	return self->firstBoxSetColumn();
}

void QVBoxPlotModelMapper_SetFirstBoxSetColumn(QVBoxPlotModelMapper* self, int firstBoxSetColumn) {
	self->setFirstBoxSetColumn(firstBoxSetColumn);
}

int QVBoxPlotModelMapper_LastBoxSetColumn(const QVBoxPlotModelMapper* self) {
	return self->lastBoxSetColumn();
}

void QVBoxPlotModelMapper_SetLastBoxSetColumn(QVBoxPlotModelMapper* self, int lastBoxSetColumn) {
	self->setLastBoxSetColumn(lastBoxSetColumn);
}

int QVBoxPlotModelMapper_FirstRow(const QVBoxPlotModelMapper* self) {
	return self->firstRow();
}

void QVBoxPlotModelMapper_SetFirstRow(QVBoxPlotModelMapper* self, int firstRow) {
	self->setFirstRow(firstRow);
}

int QVBoxPlotModelMapper_RowCount(const QVBoxPlotModelMapper* self) {
	return self->rowCount();
}

void QVBoxPlotModelMapper_SetRowCount(QVBoxPlotModelMapper* self, int rowCount) {
	self->setRowCount(rowCount);
}

void QVBoxPlotModelMapper_SeriesReplaced(QVBoxPlotModelMapper* self) {
	self->seriesReplaced();
}

void QVBoxPlotModelMapper_Connect_SeriesReplaced(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::seriesReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVBoxPlotModelMapper_ModelReplaced(QVBoxPlotModelMapper* self) {
	self->modelReplaced();
}

void QVBoxPlotModelMapper_Connect_ModelReplaced(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::modelReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVBoxPlotModelMapper_FirstBoxSetColumnChanged(QVBoxPlotModelMapper* self) {
	self->firstBoxSetColumnChanged();
}

void QVBoxPlotModelMapper_Connect_FirstBoxSetColumnChanged(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::firstBoxSetColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVBoxPlotModelMapper_LastBoxSetColumnChanged(QVBoxPlotModelMapper* self) {
	self->lastBoxSetColumnChanged();
}

void QVBoxPlotModelMapper_Connect_LastBoxSetColumnChanged(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::lastBoxSetColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVBoxPlotModelMapper_FirstRowChanged(QVBoxPlotModelMapper* self) {
	self->firstRowChanged();
}

void QVBoxPlotModelMapper_Connect_FirstRowChanged(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::firstRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVBoxPlotModelMapper_RowCountChanged(QVBoxPlotModelMapper* self) {
	self->rowCountChanged();
}

void QVBoxPlotModelMapper_Connect_RowCountChanged(QVBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVBoxPlotModelMapper*) = reinterpret_cast<void (*)(QVBoxPlotModelMapper*)>(slot);
    QVBoxPlotModelMapper::connect(self, &QVBoxPlotModelMapper::rowCountChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QVBoxPlotModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QVBoxPlotModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QVBoxPlotModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QVBoxPlotModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QVBoxPlotModelMapper_Delete(QVBoxPlotModelMapper* self) {
    delete self;
}

File: binding/include/charts/libqvboxplotmodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qvboxplotmodelmapper.html
pub const qvboxplotmodelmapper = struct {

    /// New constructs a new QVBoxPlotModelMapper object.
    pub fn New() QtC.QVBoxPlotModelMapper {
        return qtc.QVBoxPlotModelMapper_new();
    }


    /// New2 constructs a new QVBoxPlotModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QVBoxPlotModelMapper {
        return qtc.QVBoxPlotModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVBoxPlotModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvboxplotmodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, ) QtC.QAbstractItemModel {
        return qtc.QVBoxPlotModelMapper_Model(@ptrCast(self));
    }

    pub fn SetModel(self: ?*anyopaque, model: ?*anyopaque) void {
        qtc.QVBoxPlotModelMapper_SetModel(@ptrCast(self), @ptrCast(model));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QBoxPlotSeries {
        return qtc.QVBoxPlotModelMapper_Series(@ptrCast(self));
    }

    pub fn SetSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QVBoxPlotModelMapper_SetSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn FirstBoxSetColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVBoxPlotModelMapper_FirstBoxSetColumn(@ptrCast(self));
    }

    pub fn SetFirstBoxSetColumn(self: ?*anyopaque, firstBoxSetColumn: i32) void {
        qtc.QVBoxPlotModelMapper_SetFirstBoxSetColumn(@ptrCast(self), firstBoxSetColumn);
    }

    pub fn LastBoxSetColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVBoxPlotModelMapper_LastBoxSetColumn(@ptrCast(self));
    }

    pub fn SetLastBoxSetColumn(self: ?*anyopaque, lastBoxSetColumn: i32) void {
        qtc.QVBoxPlotModelMapper_SetLastBoxSetColumn(@ptrCast(self), lastBoxSetColumn);
    }

    pub fn FirstRow(self: ?*anyopaque, ) i32 {
        return qtc.QVBoxPlotModelMapper_FirstRow(@ptrCast(self));
    }

    pub fn SetFirstRow(self: ?*anyopaque, firstRow: i32) void {
        qtc.QVBoxPlotModelMapper_SetFirstRow(@ptrCast(self), firstRow);
    }

    pub fn RowCount(self: ?*anyopaque, ) i32 {
        return qtc.QVBoxPlotModelMapper_RowCount(@ptrCast(self));
    }

    pub fn SetRowCount(self: ?*anyopaque, rowCount: i32) void {
        qtc.QVBoxPlotModelMapper_SetRowCount(@ptrCast(self), rowCount);
    }

    pub fn SeriesReplaced(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_SeriesReplaced(@ptrCast(self));
    }

    pub fn OnSeriesReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_SeriesReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ModelReplaced(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_ModelReplaced(@ptrCast(self));
    }

    pub fn OnModelReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_ModelReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstBoxSetColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_FirstBoxSetColumnChanged(@ptrCast(self));
    }

    pub fn OnFirstBoxSetColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_FirstBoxSetColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LastBoxSetColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_LastBoxSetColumnChanged(@ptrCast(self));
    }

    pub fn OnLastBoxSetColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_LastBoxSetColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstRowChanged(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_FirstRowChanged(@ptrCast(self));
    }

    pub fn OnFirstRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_FirstRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn RowCountChanged(self: ?*anyopaque, ) void {
        qtc.QVBoxPlotModelMapper_RowCountChanged(@ptrCast(self));
    }

    pub fn OnRowCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVBoxPlotModelMapper_Connect_RowCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVBoxPlotModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvboxplotmodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVBoxPlotModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvboxplotmodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QVBoxPlotModelMapper_Delete(@ptrCast(self));
    }
};
